### PR TITLE
lib/runtime: deprecate legacy runtime

### LINKED ISSUE #1266: ensure execute_block works with Kusama and Polkadot blocks
## Expected Behavior

- currently execute_block is only tested with gossamer-made blocks
- need to ensure we can execute kusama/polkadot blocks
- steps: load the genesis file for the network into the state trie, get block 1 data from RPC, call execute_block with it and fix errors

related to #1210 

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by noot**
## Changes

- remove all instances of `LegacyInstance`, `LEGACY_NODE_RUNTIME`, `SUBSTRATE_TEST_RUNTIME`, `TEST_WASM` which used outdated v0.6 runtime API

## Tests

```
go test ./lib/runtime/... -short
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- related to #1268
- closes #1266

### CODE REVIEW

File: lib/runtime/wasmtime/instance.go [hunk 1]
```diff
@@ -128,7 +101,7 @@ func newLegacyInstanceFromModule(module *wasmtime.Module, engine *wasmtime.Engin
 		return nil, err
 	}
 
-	mem := instance.GetExport("memory").Memory()
+	//mem := instance.GetExport("memory").Memory()
```
Comment (arijitAD) [inline lib/runtime/wasmtime/instance.go:104]: Can this be removed?

**Review by RyRy79261 — APPROVED** [review]

File: lib/runtime/wasmer/exports.go [hunk 2]
```diff
@@ -44,28 +45,37 @@ func (in *LegacyInstance) ValidateTransaction(e types.Extrinsic) (*transaction.V
 }
 
 // Version calls runtime function Core_Version
-func (in *LegacyInstance) Version() (runtime.Version, error) {
-	version := new(runtime.LegacyVersionData)
-	ret, err := in.exec(runtime.CoreVersion, []byte{})
+func (in *Instance) Version() (runtime.Version, error) {
+	if in.version != nil && bytes.Equal(in.version.SpecName(), []byte("kusama")) {
```
[the remark this answers is not in the document]
Comment (noot) [reply lib/runtime/wasmer/exports.go:49]: done

File: lib/runtime/wasmer/instance.go [hunk 3]
```diff
@@ -302,36 +168,83 @@ func newLegacyInstance(code []byte, cfg *Config) (*LegacyInstance, error) {
...
+func (in *Instance) UpdateRuntimeCode(code []byte) error {
+	in.Stop()
+
+	imports, err := in.imports()
+	if err != nil {
+		return err
+	}
+
+	// Provide importable memory for newer runtimes
+	memory, err := wasm.NewMemory(23, 0)
```
[the remark this answers is not in the document]
Comment (noot) [reply lib/runtime/wasmer/instance.go:181]: done

File: lib/runtime/wasmtime/instance.go (hunk 1 above)
Comment (noot) [reply lib/runtime/wasmtime/instance.go:104]: removed

**Review by arijitAD — APPROVED** [review]

**Pull request merged**